lib/src/query_component.dart

```diff
@@ -189,40 +190,33 @@ class QueryComponent {
     return ReportingQueryParser.parseRevenueFromJsonString(report);
   }
 
-  List<InsertionOrder> _filterInFlightInsertionOrders(
-          List<InsertionOrder> insertionOrders) =>
-      insertionOrders
-          .where((io) =>
-              io.budget.activeBudgetSegment !=
-              InsertionOrder_Budget_BudgetSegment())
-          .toList();
-
-  DateTime _getMinStartDate(
-          List<InsertionOrder> insertionOrders) =>
-      insertionOrders
-          .map((io) => Util.convertProtoDateToDateTime(
-              io.budget.activeBudgetSegment.dateRange.startDate))
-          .reduce((date, minDate) => date.isBefore(minDate) ? date : minDate);
-
   void _addSpentToInsertionOrders(List<InsertionOrder> insertionOrders,
       Multimap<String, InsertionOrderDailySpend> spendingMap) {
     for (final io in insertionOrders) {
+      // Continues if the spending map doesn't have spending data for this io
+      // at all.
+      if (!spendingMap.containsKey(io.insertionOrderId)) continue;
```

Comment:
This case will happen when an IO has no active spending data and no historical spending data either (the IO is created recently).

Reply:
this is a great comment to leave in the code